lib/types/iprange.go

```diff
@@ -0,0 +1,262 @@
+package types
+
+import (
+	"encoding/binary"
+	"fmt"
+	"hash/fnv"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// selectMode tells how IP is selected in a IPBlock splited from --ip option
+
+type selectMode uint
+
+const (
+	roundRobin selectMode = iota
+	random
+)
+
+// IPBlock represents a continuous segment of IP addresses
+// ip     - the first IP in this address segment. CIDR can skip lower IPs
+// weight - how many IPs will be chosen from this segment. default is all
+// mode   - method to choose IP, sequential(0) or random(1). default is 0
+// hostStart - uint64 conversion of fistIP[8:16]
+// netStart  - uint64 conversion of fistIP[0:8]
+// hostN  - number of IPv4 addrs (IPv6 hosts) in this address segment
+// netN   - number of IPv6 nets in this address segment, always 1 if IPv4
+type IPBlock struct {
+	ip        net.IP
+	hostStart uint64
+	netStart  uint64
+	hostN     uint64
+	netN      uint64
+	weight    uint64
```

Comment:
Please keep this and I need it:
1. this will not impact any behaviour using this feature if an user is not aware of weight and not set weight. for example,
 --ip "1.1.1.1, 2.2.2.1-2.2.2.9, 3.3.3.0/24" will have 1+9+254 = 264 IP to be allocated sequentially to VUs.
2. My L7/L4 anti-ddos testing requires some IPs send more requests than other IPs, that means more VUs on some IPs and few VUs on other. for example, I set '--ip "1.1.1.1, 2.2.2.2|8, 3.3.3.3' to expect 80% VUs sending requests from 2.2.2.2.
3. If I set CIDRs for attacking and measuring IP for normal requests, I hope I can control CIDR's IP number to use. for example: --ip "172.16.0.0/16|10000, 192.168.1.1"